src/documentation/code-review.md

```diff
@@ -0,0 +1,63 @@
+
+## Objectives
+
+Developer and maintainer resources are scarce and every code review is an additional overhead to the actual implementation.
+An objective is, therefore, to *keep the effort for code review proportional* to the development effort.
+
+Code reviews can request changes and it is important that this feedback loop is *not obstructing the development process*.
+It must be ensured that there is both enough time for the developer to address the requested changes and that these changes are reviewed quickly.
+Also, as longer a pull request is not processed, more conflicts to other developments can occur and features can not be deployed.
+
+## Responsibilities
+
+It must be clear who is responsible to perform code reviews and these responsibilities have to be documented in the maintainer documentation.
+In research projects like Hexatomic, we propose that in general, the maintainer has this responsibility.
+They can delegate the code reviews to others if appropriate or necessary.
+E.g., if the maintainer is submitting a pull request by themselves, another person should responsible for code review.
+Delegating code reviews can be helpful to distribute the workload between several persons.
+
+In Hexatomic we have three areas where contributions require code review.
+
+### Project documentation
+
+The project documentation is similar to a formal project report.
+We, therefore, argue that ideally, the documentation should be reviewed by the Project Instructors (PIs).
+To avoid a “chicken or egg” problem, project maintainers (who are also the researches in the project) can review each other pull requests to bootstrap the initial documentation, e.g. about how code review is performed.
+
+When the project is finished, the project documentation will probably not need any updates.
+If an update is still necessary, the maintainers can decide who should review it.
+
+### User, developer and maintainer documentation
+
+The maintainer should suggest a suitable person for a code review of documentation.
+This can include other project members and PIs who are not developers but also external users.
+We argue that especially user documentation can benefit from a user perspective in the review process.
+This means that the documentation updates and the review interface must be usable for informed users.
+
+### Software code
+
+For changes to the actual Hexatomic software, a maintainer is responsible for code review.
+
+## Review schedules
+
+To find a balance between a quick code review feedback loop and a justifiable overhead for the maintainers, we propose
+to have weekly dedicated time slots where the maintainers perform the code review.
+How much time and when these time slots depend on the work schedule of the actual maintainer.
+The time slots should be documented in the `CONTRIBUTING.md` file to make them transparent to external contributors.
```

Comment:
```suggestion
The time slots should be documented in the [`CONTRIBUTING.md`](../../../CONTRIBUTING.md) file to make them transparent to external contributors.
```

Reply:
We do not need to add this to the CONTRIBUTING.md of this repository.